docs/templates/headless-cms.mdx

```diff
@@ -0,0 +1,352 @@
+
+```tsx title="headless-cms/admin/components/ContentBlocks.tsx"
+export const ContentBlocks = Component(
+	() => (
+		<BlockRepeater
+			field="blocks"
+			label={undefined}
+			discriminationField="type"
+			sortableBy="order"
+			addButtonText="Add content block"
+		>
+		...
+		/* highlight-start */
+			<Block discriminateBy="imageWithTextSection" label="Image with text section" >
+				<ContentField />
+				<ImageUploadField
+					label={locale["Image"]}
+					urlField="image.url"
+					widthField="image.width"
+					heightField="image.height"
+				>
+					<TextField field="image.alt" label={locale["Alternative text"]} />
+				</ImageUploadField>
+			</Block>
+		/* highlight-end */
+	),
+	'ContentBlocks',
+)
+```
+
+Now you can check your new section in <a href="http://localhost:1480" target="_blank">Admin</a>.
+
+#### 4. Add new section to Head
+
+Go to `headless-cms/head/components/blocks.tsx` and add new react component called `ImageWithText` which will render content and image. Next, you'll need to add your new components to block map in `Blocks` component.
+
+```tsx title="headless-cms/head/components/blocks.tsx"
+...
+/* highlight-start */
+function ImageWithText({ content, image }) {
+	return (
+		<section>
+			{content && content.parts &&
+				<RichTextRenderer blocks={content.parts} sourceField="json" />
+			}
+			<div>
+				{image &&
+					<div>
+						<img src={image.url} width={image.width} height={image.height} alt={image.alt} />
+					</div>
+				}
+			</div>
+		</section>
+	)
+}
+/* highlight-end */
+
+export default function Blocks({ blocks }) {
+	return blocks ? blocks.map((block) => {
+		const blocksElements = {
+			heroSection: <HeroSection {...block} key={block.id} />,
+			logosSection: <LogoSection {...block} key={block.id} />,
+			contentSection: <ContentSection {...block} key={block.id} />,
+			featureSection: <FeatureSection {...block} key={block.id} />,
+			ctaSection: <CtaSection {...block} key={block.id} />,
+			testimonialSection: <TestimonialSection {...block} key={block.id} />,
+			contactSection: <ContactSection {...block} key={block.id} />,
+			/* highlight-start */
+			imageWithText: <ImageWithText {...block} key={block.id} />,
+			/* highlight-end */
+		}
+
+		return blocksElements[block.type]
+	}) : null
+}
+
+```
+
+That's all. Now you can use your new section in your website.
+
+### Editting testimonials from one place
+
+You've decided that you want to edit all your testimonials in one place. You can easily do that, thanks to the Contember.
+
+#### 1. Change model
+
+Create new entities called `Testimonial` and `TestimonailAuthor` in `headless-cms/api/model/` folder.
```

Comment:
This is sooo complicated... For this example I would do something like this:

```
<MultiEditPage entities="ContentBlock[type='testimonialSection']">
  ...
</MultiEditPage>
```

Or maybe create a different "assignment"? Since what we are doing here is not just making the possible to edit from one place, but also to have "library" of them and reuse them on different pages.